### PR TITLE
CONSISTENT_FPCSR 

### PR DESCRIPTION
**Pull request by orkolorko**
## PR description
On x86_64 we switch to OpenBLAS with CONSISTENT_FPCSR flag enabled to guarantee coherent floating point rounding over all BLAS thread.
For other architectures we fall back to only one thread.

## Checklist

 [X] Updated Project.toml

## Other
I'm worried about the fact that the artifact is only available for x86_64 architectures, but is a dependency of the package.
I have no experience of this fact, so I do not know if I need to find a solution for this problem or not.

### CODE REVIEW

**Comment by lucaferranti** [comment]
Thanks! I'll have a closer look later today, but overall looks good to me.

> I'm worried about the fact that the artifact is only available for x86_64 architectures, but is a dependency of the package

This means that e.g. on mac M1 it wouldnt work? Ideally we should make this architecture dependent dependency, not sure if it's possible (I'll try to have a look).

**Comment by orkolorko** [comment]
It seems that the CI failing is independent from the PR. What should I do?

One more thing: maybe it is worth writing tests that test that effectively CONSISTENT_FPCSR does what it says;
setting this flag is what is suggested by INTLAB documentation but it would be good to have some in house tests that test if everything works as it should.

**Comment by orkolorko** [comment]
> Thanks! I'll have a closer look later today, but overall looks good to me.
> 
> > I'm worried about the fact that the artifact is only available for x86_64 architectures, but is a dependency of the package
> 
> This means that e.g. on mac M1 it wouldnt work? Ideally we should make this architecture dependent dependency, not sure if it's possible (I'll try to have a look).

It should work also for aarch64, but it failed at compilation on Yggdrasil. The compiler did not recognize the assembler codes used to switch rounding mode in the OpenBLAS source code; since I have no access to an aarch64 it is kind of difficult for me to understand the issue and I do not feel comfortable in trying to correct this in the OpenBLAS source.

**Comment by lucaferranti** [comment]
I'll think for a couple of days if I know someone who could test on aarch, otherwise we can just merge with a warning and wait to see if/when someone encounters issues

File: src/IntervalLinearAlgebra.jl [hunk 1]
```diff
@@ -42,9 +42,23 @@ include("pils/pils_solvers.jl")
...
+
 function  __init__()
     @require IntervalConstraintProgramming = "138f1668-1576-5ad7-91b9-7425abbf3153" include("linear_systems/oettli_nonlinear.jl")
     @require LazySets = "b4f0291d-fe17-52bc-9479-3d1a343d9043" include("linear_systems/oettli_linear.jl")
+    if Sys.ARCH == :x86_64
+        @info "Switching to OpenBLAS with ConsistentFPCSR = 1 flag enabled, guarantees
+        coherent floating point rounding mode over all threads"
+        BLAS.lbt_forward(OpenBLASConsistentFPCSR_jll.libopenblas_path; verbose =  true)
+    else
+        BLAS.set_num_threads(1)
```
Comment (lucaferranti) [inline src/IntervalLinearAlgebra.jl:59]: this will give rigorous results also on windows?

File: src/IntervalLinearAlgebra.jl [hunk 2]
```diff
@@ -42,9 +42,23 @@ include("pils/pils_solvers.jl")
 include("eigenvalues/interval_eigenvalues.jl")
 include("eigenvalues/verify_eigs.jl")
 
+using LinearAlgebra
+
+if Sys.ARCH == :x86_64
+    using OpenBLASConsistentFPCSR_jll
+end
```
Comment (lucaferranti) [inline src/IntervalLinearAlgebra.jl:49]: should we print a warning in the other case?

File: src/IntervalLinearAlgebra.jl [hunk 3]
```diff
@@ -42,9 +42,23 @@ include("pils/pils_solvers.jl")
...
+
 function  __init__()
     @require IntervalConstraintProgramming = "138f1668-1576-5ad7-91b9-7425abbf3153" include("linear_systems/oettli_nonlinear.jl")
     @require LazySets = "b4f0291d-fe17-52bc-9479-3d1a343d9043" include("linear_systems/oettli_linear.jl")
+    if Sys.ARCH == :x86_64
+        @info "Switching to OpenBLAS with ConsistentFPCSR = 1 flag enabled, guarantees
+        coherent floating point rounding mode over all threads"
```
Comment (lucaferranti) [inline src/IntervalLinearAlgebra.jl:56]:
```suggestion
        correct floating point rounding mode over all threads."
```

(sorry for the nitpick)

File: src/IntervalLinearAlgebra.jl (hunk 1 above)
Comment (orkolorko) [reply src/IntervalLinearAlgebra.jl:59]: I do not know; I will implement the necessary condition test for BLAS, so we can at least test this on windows.

File: src/IntervalLinearAlgebra.jl (hunk 2 above)
Comment (orkolorko) [reply src/IntervalLinearAlgebra.jl:49]: Yes, I think it is a good idea. I will review it and push another commit.

File: src/IntervalLinearAlgebra.jl (hunk 1 above)
Comment (lucaferranti) [reply src/IntervalLinearAlgebra.jl:59]: ok, we can leave the warning as it is for now, and update when we have more systematic tests

File: src/IntervalLinearAlgebra.jl (hunk 2 above)
Comment (orkolorko) [reply src/IntervalLinearAlgebra.jl:54]: I added more warnings and diagnostics, so that the user gets to know if it is in a dangerous situation

**Review by lucaferranti — APPROVED** [review]

**Pull request merged**